File: store-java/src/main/java/io/mb/store/service/external/DealerApiService.java
package io.mb.store.service.external;

import io.mb.store.resource.dto.DealerDto;
import lombok.RequiredArgsConstructor;
import org.springframework.http.MediaType;
import org.springframework.stereotype.Service;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Mono;

@Service
@RequiredArgsConstructor
public class DealerApiService {
    private static final String DEALER_BY_ID_PATH = "/dealers/{dealerId}";

    private final WebClient dealerApiWebClient;

    public Mono<DealerDto> getDealer(final String dealerId) {
        return dealerApiWebClient.get()
                .uri(it -> it.path(DEALER_BY_ID_PATH).build(dealerId))
                .accept(MediaType.APPLICATION_JSON)
                .retrieve()
                .bodyToMono(DealerDto.class);
    }
}

File: store-kotlin/src/main/kotlin/io/mb/store/service/external/DealerApiService.kt
package io.mb.store.service.external

import io.mb.store.resource.dto.DealerDto
import org.springframework.http.MediaType
import org.springframework.stereotype.Service
import org.springframework.web.reactive.function.client.WebClient
import org.springframework.web.reactive.function.client.awaitBody

@Service
class DealerApiService(
    private val dealerApiWebClient: WebClient
) {
    private companion object {
        private const val DEALER_BY_ID_PATH = "/dealers/{dealerId}"
    }

    suspend fun getDealer(dealerId: String): DealerDto {
        return dealerApiWebClient.get()
            .uri {
                it.path(DEALER_BY_ID_PATH).build(dealerId)
            }
            .accept(MediaType.APPLICATION_JSON)
            .retrieve()
            .awaitBody<DealerDto>()
    }
}